lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -2078,12 +2110,48 @@ void GrandCentralPass::runOnOperation() {
     VerbatimBuilder::Base verbatimData;
     VerbatimBuilder verbatim(verbatimData);
     verbatim += instanceSymbol;
-    auto iface =
-        traverseBundle(bundle, bundle.getID(), bundle.getPrefix(), verbatim);
+    // List of interface elements.
+    SmallVector<Operation *> interfaceElems;
+    // insertedOps will record all the interface ops and the verbatim ops
+    // inserted in the companion module, which must be erased, if the interface
+    // is redundant. The traverseBundle is a recursive traversal of the
+    // AugmentedBundleType, it will construct the Interface and populate it with
+    // the elements and insert the corresponding verbatim xmr assignments. The
+    // interface might be redundant if it is a nonlocal view and another exactly
+    // same interface is already generated. To compare the interface elements,
+    // with an existing interface, it is necessary to complete the
+    // traverseBundle traversal and gather all the elements of the interface.
+    // TODO: Update the algorithm to build the interface after traversal and not
+    // during the recursive traversal.
+    insertedOps.clear();
+    auto iface = traverseBundle(bundle, bundle.getID(), bundle.getPrefix(),
+                                verbatim, interfaceElems);
     if (!iface) {
       removalError = true;
       continue;
     }
+
+    if (companionIter.isNonlocal) {
+      // If the companion module has two exactly same ViewAnnotation.companion
+      // annotations, then add the interface for only one of them. This happens
+      // when the companion is deduped.
+      auto viewMapIter = companionToInterfaceMap.find(companionModule);
```

Comment:
Can't you fix this up when you first add the companion when you find it (much above)?